Mutagen.Bethesda.Analyzers.Skyrim/Record/Armor/VendorKeywordAnalyzer.cs

```diff
@@ -0,0 +1,97 @@
+﻿using Mutagen.Bethesda.Analyzers.SDK.Analyzers;
+using Mutagen.Bethesda.Analyzers.SDK.Results;
+using Mutagen.Bethesda.Analyzers.SDK.Topics;
+using Mutagen.Bethesda.Plugins;
+using Mutagen.Bethesda.Skyrim;
+using Noggog;
+namespace Mutagen.Bethesda.Analyzers.Skyrim.Record.Armor;
+
+public class VendorKeywordAnalyzer : IIsolatedRecordAnalyzer<IArmorGetter>
+{
+    public static readonly TopicDefinition<string, string> ArmorMissingVendorKeyword = MutagenTopicBuilder.DevelopmentTopic(
+            "Armor is missing Vendor Keyword",
+            Severity.Suggestion)
+        .WithFormatting<string, string>("Missing vendor keyword {0} but has {1}");
+
+    public IEnumerable<TopicDefinition> Topics => [ArmorMissingVendorKeyword];
+
+    private sealed record NamedFormLink(FormLink<IKeywordGetter> FormLink, string EditorID);
```

Comment:
We do have `IMajorRecordIdentifier`, but that loses the <IKeywordGetter> typing

I might add `IMajorRecordIdentifier<TMajorType> : IFormLinkGetter<TMajorType>` which could fully take `NamedFormLink`'s place
https://github.com/Mutagen-Modding/Mutagen/issues/556

Reply:
Yeah at least for this example with the form keys libraries it would be nice to access the editor as field as well although I hope this doesn't further inflate their dll size that much.

Reply:
Ah, didn't think about that.    https://github.com/Mutagen-Modding/Mutagen.Bethesda.FormKeys/issues/5